packages/desktop-client/src/components/accounts/Account.js

```diff
@@ -802,72 +885,114 @@ class AccountInternal extends PureComponent {
   };
 
   onBatchDuplicate = async ids => {
-    this.setState({ workingHard: true });
-
-    let { data } = await runQuery(
-      q('transactions')
-        .filter({ id: { $oneof: ids } })
-        .select('*')
-        .options({ splits: 'grouped' }),
-    );
+    let onConfirmDuplicate = async ids => {
+      this.setState({ workingHard: true });
 
-    let changes = {
-      added: data
-        .reduce((newTransactions, trans) => {
-          return newTransactions.concat(
-            realizeTempTransactions(ungroupTransaction(trans)),
-          );
-        }, [])
-        .map(({ sort_order, ...trans }) => ({ ...trans })),
-    };
+      let { data } = await runQuery(
+        q('transactions')
+          .filter({ id: { $oneof: ids } })
+          .select('*')
+          .options({ splits: 'grouped' }),
+      );
 
-    await send('transactions-batch-update', changes);
+      let changes = {
+        added: data
+          .reduce((newTransactions, trans) => {
+            return newTransactions.concat(
+              realizeTempTransactions(ungroupTransaction(trans)),
+            );
+          }, [])
+          .map(({ sort_order, ...trans }) => ({ ...trans })),
+      };
 
-    await this.refetchTransactions();
-  };
+      await send('transactions-batch-update', changes);
 
-  onBatchDelete = async ids => {
-    this.setState({ workingHard: true });
+      await this.refetchTransactions();
+    };
 
     let { data } = await runQuery(
       q('transactions')
-        .filter({ id: { $oneof: ids } })
+        .filter({ id: { $oneof: ids }, reconciled: true })
         .select('*')
         .options({ splits: 'grouped' }),
     );
     let transactions = ungroupTransactions(data);
 
-    let idSet = new Set(ids);
-    let changes = { deleted: [], updated: [] };
+    if (transactions.length > 0) {
+      this.props.pushModal('confirm-transaction-edit', {
+        onConfirm: () => {
+          onConfirmDuplicate(ids);
+        },
+        confirmReason: 'batchDuplicateWithReconciled',
+      });
+    } else {
+      onConfirmDuplicate(ids);
+    }
+  };
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏it seems that we perform a very similar operation multiple times throughout the code. Could we make an abstraction for it to solve the DRY issue?

Reply:
Refactored this into a shared function.